replica/table.cc

```diff
@@ -3555,8 +3573,34 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
     }
 
     tlogger.info("Cleaned up tablet {} of table {}.{} successfully.", tid, _schema->ks_name(), _schema->cf_name());
+}
+
+future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
+    auto holder = async_gate().hold();
+
+    auto* sg = storage_group_for_id(tid.value());
```

Comment:
why do it here? we're looking up the storage_group under the exclusive lock below.
the one we see there must be the authoritative one.